app/controllers/tutor_comments_controller.rb

```diff
@@ -0,0 +1,52 @@
+class TutorCommentsController < ApplicationController
+  before_action :set_tutor_comment, only: [:edit, :update, :destroy]
+
+  def new
+    @tutor_comment = TutorComment.new
+    @tutor = Tutor.find(params[:tutor])
+  end
+
+  def create
+    @tutor_comment = TutorComment.new(tutor_comment_params)
+
+    if @tutor_comment.save
+      redirect_to tutor_path(tutor_comment_params[:tutor_id])
+    else
+      @tutor = Tutor.find(tutor_comment_params[:tutor_id])
```

Comment:
so, if we add `@tutor = Tutor.find(@tutor_comment.tutor_id)` in the setter method, and render it before_action, we can get rid of all the `@tutor = ...`s.